src/connection.rs

```diff
@@ -74,6 +118,54 @@ impl Write for Connection {
     fn flush(&mut self) -> io::Result<()> {
         match self.stream {
             Stream::TcpStream(ref mut stream) => stream.flush(),
+            Stream::UdpSocket(ref mut stream) => { 
+                //udp header is 8 bytes in the begining of each datagram
+                let mut udp_header: Vec<u8> = Vec::new();
+                udp_header.write_u16::<BigEndian>(stream.request_id)?; // request id to uniquely identify response for this request
```

Comment:
I think the request id must be different **per request**, not per client. So I think we should remove the `request_id` field in the steam struct, and generate it here randomly.
https://github.com/memcached/memcached/blob/master/doc/protocol.txt#L1194-L1201

Reply:
I recommend keeping it the same as the document recommends too. I also see in libmemcached (client/ms_conn.c), request id is incremented for each request.